openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -202,7 +206,8 @@ def get_cohort(user, course_key, assign=True, use_cached=False):
             assignment_type=CourseCohort.RANDOM
         ).course_user_group
 
-    user.course_groups.add(cohort)
+    membership = CohortMembership(course_user_group=cohort, user=user)
```

Comment:
With CohortMembership existing, should user.course_groups change (be deleted or changed to redirect to CohortMembership?). Same question about cohort.users.